src/Sarif.Viewer.VisualStudio/Tags/SarifLocationTagger.cs

```diff
@@ -0,0 +1,573 @@
+            {
+                foreach (var possibleTag in possibleTags.Where((possibleTag) => possibleTag.DocumentPersistentSpan.Span != null))
+                {
+                    SnapshotSpan possibleTagSnapshotSpan = possibleTag.DocumentPersistentSpan.Span.GetSpan(span.Snapshot);
+                    if (span.IntersectsWith(possibleTagSnapshotSpan))
+                    {
+                        yield return new TagSpan<TextMarkerTag>(possibleTagSnapshotSpan, possibleTag.Tag);
+                    }
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public IDisposable Update()
+        {
+            return new BatchUpdate(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposed)
+            {
+                return;
+            }
+
+            disposed = true;
+
+            if (disposing)
+            {
+                using (TagListLock.EnterReadLock())
+                {
+                    // Important note that we do not dispose or clear
+                    // the SARIF tag list as that would destroy the whole purpose of using
+                    // "persistent spans" which survive open and close of a document within
+                    // a VS session.
+                    foreach (var sarifTag in this.sarifTags)
+                    {
+                        sarifTag.PropertyChanged -= this.SarifTagPropertyChanged;
+                    }
+                }
+
+                using (SarifTaggersLock.EnterWriteLock())
+                {
+                    SarifTaggers.Remove(this);
+                }
+
+                this.batchUpdateLock.InnerLock.Dispose();
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void SarifTagPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (sender is SarifLocationTag sarifTag)
+            {
+                using (this.Update())
+                {
+                    this.UpdateBatchSpan(sarifTag.DocumentPersistentSpan.Span);
+                }
+            }
+        }
+
+        private void UpdateBatchSpan(ITrackingSpan snapshotSpan)
+        {
+            // If there currently is a batch span, update it to include the biggest
+            // range of buffer affected so far.
+            if (this.batchUpdateSpan == null)
+            {
+                this.batchUpdateSpan = snapshotSpan;
+                return;
+            }
+
+            ITextSnapshot snapshot = this.textBuffer.CurrentSnapshot;
+
+            SnapshotSpan currentBatchSpan = this.batchUpdateSpan.GetSpan(snapshot);
+            SnapshotSpan currentUpdate = snapshotSpan.GetSpan(snapshot);
+
+            SnapshotPoint newStart = currentBatchSpan.Start.Position < currentUpdate.Start.Position ? currentBatchSpan.Start : currentUpdate.Start;
```

Comment:
>Position [](start = 60, length = 8)

Math.Min, Max

Reply:
Cannot use Math here because < operator is overloaded and Math doesn't deal with that.

---
In reply to: [482259488](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r482259488) [](ancestors = 482259488)